### PR TITLE
Fix: Replace `sqllineage` with `openmetadata-sqllineage`

### PR DESCRIPTION
**Pull request by nahuelverdugo**
### Describe your changes :
I have updated the `sqllineage` used until now with `openmetadata-sqllineage`, which adds an option to extract lineage by using `sqlfluff` as the parser and considers the dialect of the query. Since it is a recent change, we will use the `sqlfluff` parser option as a fallback for the usual lineage extraction. 

In addition, the `LineageParser` now supports the dialect parameter to use `sqllineage` with `sqlfluff`.

### Type of change :
- [x] Improvement

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/lineage/models.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    SNOWFLAKE = "snowflake"
+    SOQL = "soql"
+    SPARKSQL = "sparksql"
+    SQLITE = "sqlite"
+    TERADATA = "teradata"
+    TSQL = "tsql"
+
+
+MAP_CONNECTION_TYPE_DIALECT: Dict[str, Dialect] = {
+    str(MySQLType.Mysql.value): Dialect.ATHENA,
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/lineage/models.py:79]: could we add a comment on why this mapping?

Comment (nahuelverdugo) [reply ingestion/src/metadata/ingestion/lineage/models.py:79]: Moved the logic into `ConnectionTypeDialectMapper` class.

**Review by pmbrull — DISMISSED** [review]

**Comment by akash-jain-10** [comment]
Hello @nahuelverdugo . How are we publishing the `openmetadata-sqllineage` python package ? Is that something we need to do for every release ? Can you please share some details for the same here. Thanks.

**Comment by nahuelverdugo** [comment]
> Hello @nahuelverdugo . How are we publishing the `openmetadata-sqllineage` python package ? Is that something we need to do for every release ? Can you please share some details for the same here. Thanks.

We have created a repo for that https://github.com/open-metadata/openmetadata-sqllineage, and there is an action that releases when a release is created. It is a fork of the `sqllineage` library that we did to start testing a change done in the used parser for the queries. We don't need to do it on each release, and we hope the repo is temporary until we merge https://github.com/reata/sqllineage/pull/326.

**Review by akash-jain-10 — APPROVED** [review]

**Review by OnkarVO7 — APPROVED** [review]

**Pull request merged**